src/storage/tuple.cpp

```diff
@@ -59,6 +59,10 @@ void Tuple::SetValue(const oid_t column_offset, const type::Value &value,
 
   // Skip casting if type is same
   if (type == value.GetTypeId()) {
+    if ((type == type::Type::VARCHAR || type == type::Type::VARBINARY)
+        && value.GetLength() > column_length) {
+      throw peloton::IncompatibleTypeException(type, " data is too long");
```

Comment:
Postgres reports an ERRCODE_STRING_DATA_RIGHT_TRUNCATION in such situation. Should we add a new Exception for this? 
Is ValueOutOfRangeException or CastException more suitable here?